### PR TITLE
Update to @codingame/moanco-vscode-api version 2

### PR DESCRIPTION
**Pull request by kaisalmen**
Everythings works again expect  for `monaco-editor-workers` have been removed. This will be fixed with upcoming merge of monaco-editor-wrapper and the example alignment.

It already uses `@codingame/esbuild-import-meta-url-plugin`
I aligned the lint configuration with other TypeFox projects (the delta is not huge).

I will fully update the README when all other work is completed.

### CODE REVIEW

**Comment by kaisalmen** [comment]
@CGNonofr I would like to continue the wrapper / example merge with this preview release and upgrade whenever there is something new.

**Comment by kaisalmen** [comment]
@CGNonofr this is ready now. All doc updates will come later after the wrapper/example merge.

File: README.md [hunk 1]
```diff
@@ -85,16 +87,14 @@ Earlier in 2023 we started to treemend an existing `monaco-editor` dependency vi
...
 ```yaml
   "overrides": {
-    "monaco-editor": "npm:@codingame/monaco-editor-treemended@>=1.85.5 <1.86.0",
-    "vscode": "npm:@codingame/monaco-vscode-api@>=1.85.5 <1.86.0"
+    "monaco-editor": "npm:@codingame/monaco-vscode-editor-api@>=2.0.0 <2.1.0"
   },
   "resolutions": {
-    "monaco-editor": "npm:@codingame/monaco-editor-treemended@>=1.85.5 <1.86.0",
-    "vscode": "npm:@codingame/monaco-vscode-api@>=1.85.5 <1.86.0"
+    "monaco-editor": "npm:@codingame/monaco-vscode-editor-api@>=2.0.0 <2.1.0"
```
Comment (CGNonofr) [inline README.md:97]: why removing the vscode override?

File: packages/client/package.json [hunk 2]
```diff
@@ -48,18 +48,27 @@
...
-    "@codingame/monaco-vscode-editor-service-override": "~1.85.5",
-    "@codingame/monaco-vscode-languages-service-override": "~1.85.5",
-    "@codingame/monaco-vscode-model-service-override": "~1.85.5",
-    "monaco-editor": "npm:@codingame/monaco-editor-treemended@>=1.85.5 <1.86.0",
-    "vscode": "npm:@codingame/monaco-vscode-api@>=1.85.5 <1.86.0",
+    "@codingame/monaco-vscode-editor-service-override": "~2.0.0",
+    "@codingame/monaco-vscode-languages-service-override": "~2.0.0",
+    "@codingame/monaco-vscode-model-service-override": "~2.0.0",
+    "@codingame/monaco-vscode-api": "~2.0.0",
+    "@codingame/monaco-vscode-editor-api": "~2.0.0",
```
[the remark this answers is not in the document]
Comment (kaisalmen) [reply packages/client/package.json:55]: `~2.0.0` is equivalent. The notation with the tilde `~` did not work in the past when using  `npm:..` syntax that's why we have both. I would like to write only `~2.0.0`. Maybe this changed with npm 10. Have to check it.

[the remark this answers is not in the document]
Comment (CGNonofr) [reply packages/client/package.json:55]: Oh I didn't remember it, thanks

File: README.md (hunk 1 above)
Comment (kaisalmen) [reply README.md:97]: I came to the conclusion it is not needed. The dependency is declared and there usually is no mismatching package.

**Review by CGNonofr — APPROVED** [review]
LGTM then :+1:

**Pull request merged**